### ISSUE TITLE
Hang after screen unlock

### ISSUE THREAD

**Issue by FREEscanRIP** [M66B/NetGuard#425]
After keeping NetGuard enabled for a while, every time I unlock my phone and try to access my Recent apps list, it hangs for a good 5 (sometimes 10) seconds. Screenshot: https://imgur.com/5OPDQ70

**Comment by M66B** [comment]
How do you know this is being caused by NetGuard?
Can you provide a log file to look at?

**Comment by licaon-kter** [comment]
Xperia something, model? Android version?

NetGuard->Menu->About->tab dialogue 7 times->save logcat.txt->use https://gist.github.com to paste the log contents (do look at the log first for sensitive info) and share the link here.

**Comment by FREEscanRIP** [comment]
I'm using the stock 5.1.1 Android on my Xperia Z2. It's not rooted.

The issue never happened before installing NetGuard. Also, if I force close NetGuard, the problem disappears. But as soon as I open it, even without enabling it, it starts doing it again.

I attached that screenshot because it clearly shows how much CPU NetGuard is using for ~10 seconds after I turn the screen on (I don't even have to unlock it).

The logcat.txt: https://gist.github.com/FREEscanRIP/a303fa38e81320c879e30bde59a31dc5

**Comment by M66B** [comment]
On screen on/off the rules are applied when the conditions are different.
NetGuard requests Android to apply the new rule set.
Android does the processing, not NetGuard, but the CPU is accounted to NetGuard.
In other words this cannot be improved.
On my Nexus 5X there is no delay, so either your CPU is weak or the CPU governor of the kernel is badly tuned.

**Comment by M66B** [comment]
Can you please try if this version improves things:
https://github.com/M66B/NetGuard/releases/tag/2.24

**Comment by FREEscanRIP** [comment]
After installing v2.24, nothing has changed. I tweaked the settings, I even set "Delay screen off" to a huge number, just to (kind of) disable the screen-off rules, but the delay stayed the same.
I decided to Force close NetGuard, run Greenify (not on NetGuard), clear my Recent apps list and reboot. After that, I re-enabled NetGuard. The delay has been reduced to a second or two.
After that, I downgraded to v2.23, just to test things. The delay stayed at 1-2 seconds. It might seem like that the reboot solved the problem, but I did reboot my phone before, and the problem reappeared eventually.
Right now, with the delay of 1-2 seconds, it seems to be working fine. I'll keep using v2.24 and I'll notify you if the problem reappears.
Also, as you said, it might be a firmware problem. I should get Marshmallow soonish, so that might solve this completely.
